src/aws_encryption_sdk/internal/crypto/authentication.py

```diff
@@ -47,11 +46,10 @@ def __init__(self, algorithm, key):
 
     def _set_signature_type(self):
         """Ensures that the algorithm signature type is a known type and sets a reference value."""
-        try:
-            verify_interface(ec.EllipticCurve, self.algorithm.signing_algorithm_info)
-            return ec.EllipticCurve
-        except InterfaceNotImplemented:
-            raise NotSupportedError("Unsupported signing algorithm info")
+        for method in ec.EllipticCurve.__abstractmethods__:
+            if not hasattr(self.algorithm.signing_algorithm_info(), method):
+                raise NotSupportedError("Unsupported signing algorithm info")
```

Comment:
Issue/Blocking: You MUST NOT copy and paste code.
Abstract and simplify: author `verify_interface` in 
`src/aws_encryption_sdk/internal/utils/__init__.py`.

Reference it here and in `src/aws_encryption_sdk/internal/crypto/elliptic_curve.py`.

Reply:
makes sense! i'll refactor this out into a function. I missed that. I guess I was happy that I was able to get it to work that I did not think of that. Good call out!